lib/modules/manager/gitlabci/extract.spec.ts

```diff
@@ -155,5 +160,74 @@ describe('modules/manager/gitlabci/extract', () => {
         },
       ]);
     });
+    it('extracts from image', () => {
+      let expectedRes = {
+        autoReplaceStringTemplate:
+          '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}',
+        currentDigest: undefined,
+        currentValue: 'test',
+        datasource: 'docker',
+        depName: 'image',
+        depType: 'image',
+        replaceString: 'image:test',
+      };
+
+      expect(extractFromImage('image:test')).toEqual(expectedRes);
+
+      expectedRes = { ...expectedRes, depType: 'image-name' };
+      expect(
+        extractFromImage({
+          name: 'image:test',
+        })
+      ).toEqual(expectedRes);
+
+      expect(extractFromImage(undefined) === undefined).toBeTruthy();
+    });
+
+    it('extracts from services', () => {
+      const expectedRes = [
+        {
+          autoReplaceStringTemplate:
+            '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}',
+          currentDigest: undefined,
+          currentValue: 'test',
+          datasource: 'docker',
+          depName: 'image',
+          depType: 'service-image',
+          replaceString: 'image:test',
+        },
+        {
+          autoReplaceStringTemplate:
+            '{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}',
+          currentDigest: undefined,
+          currentValue: 'test2',
+          datasource: 'docker',
+          depName: 'image2',
+          depType: 'service-image',
+          replaceString: 'image2:test2',
+        },
+      ];
+      const services = ['image:test', 'image2:test2'];
+      expect(extractFromServices(undefined) === undefined).toBeTruthy();
```

Comment:
```suggestion
      expect(extractFromServices(undefined)).toUndefined();
```
use simpler expectations.